WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java

```diff
@@ -541,16 +541,22 @@ public int compare(SiteRecord site1, SiteRecord site2) {
             }
 
             if (mSites == null || !mSites.isSameList(sites)) {
-                mAllSites = (SiteList) sites.clone();
-                mSites = filteredSitesByTextIfInSearchMode(sites);
+                SiteList allSites = (SiteList) sites.clone();
+                SiteList filteredSites = filteredSitesByTextIfInSearchMode(sites);
+
+                return new SiteList[]{allSites, filteredSites};
             }
 
             return null;
         }
 
         @Override
-        protected void onPostExecute(Void results) {
-            notifyDataSetChanged();
+        protected void onPostExecute(SiteList[] updatedSiteLists) {
+            if (updatedSiteLists != null) {
```

Comment:
Using a `synchronized` block would be an extreme measure without real value in this case. Problem before was that the instance variable was changed too early (and in a background thread) and the recycler view had the opportunity to repaint its children (for whatever reason) before the `notifyDatasetChanged` call was issued.

Now that the instance variable change and the `notify` both happen on the main thread (and close together) the adapter and the RecyclerView don't have a chance to observe out-of-date data. The `synchronized` block will add no value since there are only 2 code points that `mSites` is changed and both now happen on the main thread. So, `synchronized` would only confuse the reader at this stage.